### DISCUSSION TITLE
IBM i Renderer - Rulers or ability to reference item position

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#983] [Ideas]
I know that `IBM i Renderer` is still very fresh but I use it almost daily.   A significant amount recently. 

One item I have issues with is determining where I need to move an existing item on screen.  I end up typing in column values until its just what I need.

I think it would make my effort easier if I could have rulers for rows and columns or be able to get a grid position by clicking on the render view. 

Is this something that would be easy to incorporate?

**Comment by worksofliam** [comment]
Perhaps. It'd be even better if we just had a drag and drop situation I think. Let me see how we could get this to work.

**Comment by m-tyler** [comment]
I agree about how nice drag and drop would be but if I could prioritize the needs for this extension I would have field sorting, a ruler, an item position tool tip, then field editing (to show the best estimate of on-screen size).  

I appreciate the effort in this tool it keeps me from needing to use RDi or SDA (mostly).

**Reply by worksofliam** [reply]
Can you describe some of these for me? Not sure I know what their functionality would be.

* field sorting
* item position tool tip

**Comment by m-tyler** [comment]
Field Sorting:
In SDA, option 12=Design image, F4 work with Fields, F6 Sort by row/column.  SDA and RDI sort differently when it comes to hidden and program-to-program fields, but I am not concerned with which sort option is chosen.
 * RDI sort
![image](https://user-images.githubusercontent.com/1353665/207634207-d1949027-f1c8-435a-a612-9f1f62c47f9c.png)
 * SDA sort
![image](https://user-images.githubusercontent.com/1353665/207636229-66b4369d-3415-492f-8550-9647cb81172f.png)



Item position tool tip:
This is only found in RDi DSPF designer.  You hover/click on an item and you can see where on the screen its start row/column is.   
![image](https://user-images.githubusercontent.com/1353665/207634041-00a2e699-8a8d-4978-8564-3a05710582e1.png)


Thanks for the considerations!!!

**Reply by worksofliam** [reply]
I can't promise the sort is going to be added, but I did put the position in the tooltip: 

<img width="677" alt="image" src="https://user-images.githubusercontent.com/3708366/207657622-264409cf-c063-4178-8b07-ad94b55022ab.png">

**Reply by m-tyler** [reply]
Well then just ignore my message below about not doing to tool tip !😀😀😀

**Reply by worksofliam** [reply]
It was an easy turn around :) Will release it later this week

**Comment by m-tyler** [comment]
Actually, you can drop the consideration of the tooltip hover to show the row and column or at least make a really low pty enhancement.   I think having a ruler is far better if I could get any of the enhancements I have mentioned (outside of getting drag and drop).  

Again, thanks for the work you put into this extension it really helps work with DSPFs.
